cpp-ch/local-engine/Functions/SparkFunctionGetJsonObject.h

```diff
@@ -679,15 +720,63 @@ class FlattenJSONStringOnRequiredFunction : public DB::IFunction
         return is_doc_ok;
     }
 
+private:
+    DB::ContextPtr context;
+    /// If too many rows cannot be parsed by simdjson directly, we will normalize the json text at first;
+    mutable bool is_most_normal_json_text = true;
+    mutable size_t total_parsed_rows = 0;
+    mutable size_t total_normalized_rows = 0;
+
+    template<typename JSONParser, typename JSONStringSerializer>
+    void insertResultToColumn(
```

Comment:
It seems to be complex, I guess there should be a simpler implement with less branches

Reply:
Should explain whic case it is for each branch